<h1 id="introduction">Introduction</h1>
<ul>
<li>Most proteomic studies are based on mass spectrometry</li>
<li>The program will be finished in December, in January we will only do exercises</li>
<li>The examination is composed in a poster presentation and an oral part
<ul>
<li>A poster is a self-sufficient work</li>
<li>Usually it is prepared at the beginning of a scientistâ€™s career</li>
<li>We should base our poster on selected papers</li>
<li>The poster is not mandatory, but if we do it we will only be asked 1 question in the exam</li>
</ul></li>
<li>Proteomics is one of the main omics, but also interactomics and secretomics are becaming really interesting</li>
<li>The data produced by omics are on the order of zetta and yottabytes</li>
<li>Personalized medicine is important, and now we have the tools to implement it</li>
<li>Proteomics can be structural, functional and differential (expression)</li>
<li>Three main approaches: top-down, bottom-up and shotgun</li>
<li>Many funding sources now require researchers to share raw data</li>
<li>In silico-cloning: we need to know at least one software</li>
<li>Other useful softwares are Pymol, Graphpad Prism, ImageJ
<ul>
<li>Graphpad Prism is a spreadsheet used to elaborate data and do statistics</li>
</ul></li>
<li>Proteomics is the large scale study of proteins, the proteome is the total amount of proteins coded in the genome</li>
<li>Proteomics involves a spatial and temporal dimension, while the genome is static
<ul>
<li>It is really noisy (!)</li>
<li>Proteins are really diverse, and their processing is complex</li>
<li>They are present in a really broad concentration range</li>
<li>There is no PCR for proteins (!)</li>
</ul></li>
<li>Unnatural amino acids are really interesting, and there are many patens about them
<ul>
<li>They are used frequently in cosmetics (!)</li>
</ul></li>
<li>Proteases are everywhere (!)</li>
<li>Aromatic amino acid tend to be in  <img class="inlinemath" src="eqn000.png" WIDTH=12 HEIGHT=19 STYLE="vertical-align: -4px; margin: 0;" alt="\beta" /> conformations</li><li>MALEK tend to be in  <img class="inlinemath" src="eqn001.png" WIDTH=12 HEIGHT=10 STYLE="vertical-align: -1px; margin: 0;" alt="\alpha" /> helices</li><li>1/3 of drugs target GPCR receptors</li>
<li>Membrane proteins are hard to study</li>
<li>An antibody is around 150 kDa, it is a big protein</li>
<li>Is of huge clinical interest to find fluorescent proteins that emit in the red spectrum
<ul>
<li>Most tissues absorb really well green light, and red is the least absorbed part of the spectrum</li>
</ul></li>
<li>Mutagenesis can be performed rationally or randomly
<ul>
<li>Random mutagenesis is made with kits that make an imprecise PCR</li>
<li>We can substitute any possible aminoacid in a specific postion that we know to be important</li>
</ul></li>
<li>The top-down aprroach starts from the protein</li>
<li>The bottom-up approach uses peptides deriving form digestion of the protein</li>
<li>The shotgun approach uses different peptides deriving from many proteins</li>
<li>The less complex the protein mixture, the better (!)</li>
<li>Informations on chemical compounds can be retrieved in PubChem</li>
<li>PubMed does not indicize articles outside of biology and medicine (!)
<ul>
<li>Better to use Scopus or Web of Science for chemistry and other fileds</li>
<li>For patens, Espacenet is the main database</li>
</ul></li>
<li>A volcano plot has the fold change in expression in the x axis and the significance in the y axis
<ul>
<li>It is used to rapidly identify differentially expressed proteins (it is also applicable to other fields)</li>
</ul></li>
</ul>
<h1 id="random-notes">Random notes</h1>
<ul>
<li>Bioluminescence seems that appeared as a way to get rid of oxygen, when it firs appeared in geological times</li>
</ul>
<h1 id="protein-separation">Protein separation</h1>
<ul>
<li>Proteins can be separed with different chromatographies or electrophoresis, microdialysis</li>
<li>We can use depletion techniques to remove typical high-abbundance proteins
<ul>
<li>There are kits to remove albumin with spin columns</li>
</ul></li>
<li>Protein electrophoresis can be done in native conformation or by SDS-PAGE</li>
<li>Immunological techniques are Western blot, ELISA, immunoprecipitation
<ul>
<li>Antibodies have affinity constants ranging from  <img class="inlinemath" src="eqn002.png" WIDTH=24 HEIGHT=17 STYLE="vertical-align: -1px; margin: 0;" alt="10^9" /> to  <img class="inlinemath" src="eqn003.png" WIDTH=33 HEIGHT=17 STYLE="vertical-align: -1px; margin: 0;" alt="10^12" /> M</li></ul></li>
<li>Indirect ELISA is more sensitive than the direct version
<ul>
<li>A primary Ab has several epitopes (!)</li>
</ul></li>
<li>Sandwich ELISA is more specific because requires the contemporaneous recognition of 2 non-overlapping epitopes</li>
<li>Competitive ELISA is more senitive and requires less Ab
<ul>
<li>It is basically a negtaive version, where if the Ab is washed away means that there was antigen in the sample</li>
</ul></li>
<li>Sample solubilization is done in Urea 7M, Thiourea 2M and Chaps 4%</li>
<li>SS bridges can be broken with DTT or tributilphosphine</li>
<li>SDS-PAGE is stained with Comassie or silver staining</li>
<li>The upper limit for the number of resolvable proteins in a 2D-PAGE is 10000</li>
<li>Spot identification in 2D-PAGE can be done by MS or by image analysis
<ul>
<li>Image analysis can be done by comparing our image with gel images of identified proteins present in databases</li>
<li>I usually work on master images created by combining 5-10 images of gels run in the same condition, to reduce noise</li>
</ul></li>
<li>Databases for 2D-PAGE, like ExPasy proteomic server, allow to search for images of specific tissues or cells, or conditions
<ul>
<li>I can then compare my spots with the ones already identified</li>
</ul></li>
<li>Differential PAGE (DIGE) runs 2 samples on the same gel, with different labels, so to be able to compare expression differences between them
<ul>
<li>It is a good practice to repeat the experiment by switching labels</li>
<li>Labels called cyainine derivatives (CyDies) are available with many different emission spectra
<ul>
<li>Labelling is done prior to IEF</li>
<li>We can use many samples at the same time (multiplexing)</li>
<li>They are really small and bind to the  <img class="inlinemath" src="eqn004.png" WIDTH=9 HEIGHT=10 STYLE="vertical-align: -1px; margin: 0;" alt="\epsilon" /> amino group of Lys in proteins</li></ul></li>
</ul></li>
<li>The staining method for 2D-PAGE should be chosen considering the sensitivity required</li>
<li>2D-PAGE is generally speaking non-quantitative</li>
<li>We can perform WB also on 2D-PAGE (!)</li>
<li>HILC (Hydrophilic interaction liquid chromatography) uses a polar stationary phase and a polar mobile phase
<ul>
<li>It is really useful for glycosilated proteins</li>
<li>Elution is done in water gradient</li>
</ul></li>
<li>N-terminal microsequencing is based on Edman degradation and can be used for spot indentification, but it is obsolete
<ul>
<li>It has been replaced by MS</li>
<li>It uses phenyl-isothiocyanate that reacts with the N-termini forming thiazolinone derivatives</li>
<li>The derivative is then released and identified by cromatography or electrophoreys, before repeating the cycle</li>
<li>Amino acids are red at 269 nm</li>
</ul></li>
</ul>
<h1 id="mass-spectrometry---basis-is-not-for-exam">Mass spectrometry - basis is not for exam</h1>
<ul>
<li>A mass spectrometer contains an ion source, a mass filter and a detector</li>
<li>Some ion sources can be interfaced with a separation technique, others must be operated manually</li>
<li>Soft ionization method do not break chemical bonds, while hard methods do that
<ul>
<li>In proteomics we use soft ionization: MALDI or ESI</li>
<li>There is also a semi-soft technique: fast atom bombardment (FAB)
<ul>
<li>It is used for small proteins and peptides</li>
<li>It uses typically a glycerol matrix</li>
</ul></li>
<li>In ionization techniques, the matrix is used in order to prevent sample fragmentation</li>
</ul></li>
<li>In ESI the sample is in solution and it is sprayed by a small tube in a really strong electric field in presence of a hot nitrogen flow
<ul>
<li>The ionization is performed at atmosferic pressure</li>
<li>The sample forms droplets that evaporate concentrating charge, until they undergo columbic explosion</li>
<li>The produce ions are multiple-charged, and this is useful for detecting big molecules</li>
<li>We can operate both in positive or negative ionization
<ul>
<li>In positive mode I need to add formic acid to the solvent, in negative mode ammonia</li>
<li>If I have acidic groups in my analite I want to operate in negative modality, while with basic groups I want to operate in positive modality</li>
</ul></li>
<li>It is possible to form protonated ions, deprotonated ions or cation adducts
<ul>
<li>It is very sensitive to salts and detergents</li>
</ul></li>
<li>Since I have multiple-charged ions, I have many peaks for each analite
<ul>
<li>With molecules under 1200 Da I tend to have single ions</li>
<li>The multiple peaks are normally distributed and can be deconvoluted via software to derive the MW of the ion
<ul>
<li>This is done by transformation or maximum entropy</li>
</ul></li>
</ul></li>
<li>Native and complexed proteins tend to have non-normal distribution of ions, while denatured proteins have a gaussian peak distribution</li>
<li>It is very sensitive, but I need really pure samples</li>
</ul></li>
<li>In MALDI the solid sample is heated by a laser while embedded in the matrix
<ul>
<li>The matrix becomes ionized and transfer charge to the analite, while exploding at supersonic speed</li>
<li>We can operate in negative or positive modality</li>
<li>It generates single-charged ions M+H and M-H</li>
<li>It is more tolerant to contaminats</li>
<li>It can directly analize cells (!)</li>
<li>MALDI resolution can be influenced by the differential time of ionization of analite molecules, and differential initial velocity of the ions before the TOF acceleration</li>
</ul></li>
<li>A mass spectrum is a plot with m/z on the x axis and relative abundance in the y axis
<ul>
<li>Peaks are typically really sharp</li>
<li>From each ion I get a carachteristic pattern of peaks, due to spontaneous fragmentation</li>
<li>I have spontaneous fragmentation only if my ionizing source is strongh enough</li>
</ul></li>
<li>The accuracy of the mass spectrometer is measured in ppm</li>
<li>The resolution of a mass specrtrometer is obtained by dividing height of the peak by its width at half height</li>
<li>A mass filter is a device that separates analite ions based on their m/z</li>
<li>A magnetic sensor mass analyzer (MSA) deflect the ions by differen radii depending on their m/z</li>
<li>The quadrupole mass filter allows a stable path only for a selected m/z, and we can get a spectrum by scanning all the target m/z values
<ul>
<li>It is used also in tandem MS</li>
</ul></li>
<li>Orbitrap (Makarov, 2000) is the newest and most sensitive analyzer
<ul>
<li>There is a spindle-like electrode around which the analite ions spin and a barrel-like electrode</li>
<li>The ions oscillate with a frequency related to their m/z, giving a signal</li>
<li>The signal is deconvoluted with the Fourier transform to yeld the fundamental armonics, which are related to m/z</li>
</ul></li>
<li>The time of flight analyzer (TOF) accelarates ions with an electric field in a way that is proportional to m/z
<ul>
<li>The time that is required for the ions to reach the detector is proportional to m/z</li>
</ul></li>
<li>The reflectron is a variant of TOF that improve resolution by compensating for different initial velocities</li>
<li>MS detectors are diods that generate secondary electrons when struck by an ion
<ul>
<li>Electron multipliers produce many secondary electron for each ion. and therefore are more sensitive, but require extensive maintenance</li>
</ul></li>
<li>Tandem MS combines 2 MS in the same instrument
<ul>
<li>I avoid fragmentation in the first step using a soft ionization of the molecular ion</li>
<li>A collision cell between the 2 MS filled with inert gas allows to fragment the analite</li>
<li>Fragmentation happens because of CAD (collision activated) and CID (collision induced)</li>
</ul></li>
</ul>
<h1 id="peptide-mass-fingerprinting">Peptide mass fingerprinting</h1>
<ul>
<li>It is the main bottom-up approach</li>
<li>The standard workflow starts with 2D-PAGE that allows to recover unique spots</li>
<li>It is important to chose a staining method that is compatible with MS</li>
<li>We can also use multidimensional HPLC as an alternative to 2D-PAGE</li>
<li>Spots are then cut and destained</li>
<li>Spot picking can be done manually or with a robot
<ul>
<li>Manual picking is susceptible to keratin contamination</li>
<li>There is a risk for gel deformation</li>
</ul></li>
<li>The protein is then digested by trypsin to yeld peptides
<ul>
<li>Trypsin cuts after K or R, but only if not followed by P</li>
<li>It is important to have a complete digestion to avoid missing cleavage sites</li>
<li>Better to use volatile buffers to eliminate them easily afterwards</li>
<li>Trypsin can also self-digest (!) and the deriving peptides are really useful as a standard for MS
<ul>
<li>It is an internal calibrator</li>
</ul></li>
</ul></li>
<li>Peptides are then purified by reverse chromatography or Zip tips
<ul>
<li>RC uses apolar stationary phase on polar mobile phase</li>
<li>Zip tips are a miniature RC column (!)</li>
</ul></li>
<li>We perform MS on the peptides, getting a fingerprint of the protein
<ul>
<li>In MS peptides must be ionized in a gas phase</li>
<li>MS measures the m/z ratio of the peptide ions</li>
</ul></li>
<li>I can then identify the protein by searching for my fingerprint in databases</li>
<li>Masses can be reported as monoisotopic or average
<ul>
<li>Entry level instrument cannot differentiate isotopes, therefore report only average masses</li>
</ul></li>
<li>To match a fingerprint with a database, I check how many hits on the same protein I have with my masses
<ul>
<li>I choose the protein with the maximum number of hits</li>
</ul></li>
<li>Main PMF databases are Mascot and ProFound</li>
</ul>
<h1 id="peptide-de-novo-sequencing">Peptide de novo sequencing</h1>
<ul>
<li>Peptide de novo sequencing uses MS/MS spectra to determine the sequence of a protein without using any previous knowledge</li>
<li>This is in contrast with database search, that identifies the peptide using databases</li>
<li>The fragmentation process produces different kinds of ions
<ul>
<li>Give the low collision energy employed, most fragmentations involve peptide bonds</li>
<li>If the charge is retained in the N-terminal fragment, the ion is termed a, b or c</li>
<li>If the charge is retained in the C-terminal fragment, the ion is termed x, y or z</li>
<li>Fragmentation of the peptide bond produces y or b fragments</li>
<li>The nomenclature of the fragments derives from Ropestroff and Foleman</li>
</ul></li>
<li>Fragmentation can be done by CID or ETD (electron transfer)
<ul>
<li>CID produces mostly b and y ions because the collision energy is low and can fragment only the peptide bond</li>
</ul></li>
<li>I can recover the mass of the residues by analyzing the mass difference between ions
<ul>
<li>To do this I need the entire b or y series (!)</li>
<li>The peptide mass is given by the b-OH (-17u) or by y+H+ (-1u)</li>
</ul></li>
<li>Given a MS/MS spectrum, the sofware Peaks can give the protein sequence, with a confidence for each residue</li>
<li>Peaks gives the statistic ALC (average local confidence, the average likelyhood of an aa in the sequence) and a global statistic (TLC, the expected number of correct aa) for the peptide</li>
</ul>
<h1 id="synthetic-biology---not-for-exam">Synthetic biology - Not for exam</h1>
<ul>
<li>In order to provide proprieties that are not available in nature we can use non-natural amino acids</li>
<li>We can also modify polymerases in order to use non-natural nucleotides that still pair among themselves in a specific way</li>
</ul>
<h1 id="molecular-cloning---not-for-exam">Molecular cloning - Not for exam</h1>
<ul>
<li>Molecular cloning is essential because it is our only way to obtain high amounts of proteins</li>
<li>The general workflow is to isolate the cDNA of the protein, insert it in a plasmid and transfect bacteria with it
<ul>
<li>The cDNA is amplified with primers containing a 3â€™ overhang that allows to introduce the appropriate restriction sites</li>
</ul></li>
<li>When I cut my plasmid, I want to use single cutters so to avoid that the plasmid could close on itself
<ul>
<li>Usually I can cut exactly where I want because plasmids are engineered to have many restriction sites</li>
</ul></li>
<li>Selection of recombinants is done with antibioitics for bacteria and mammalian cells, and with auxotrofs for yeast</li>
<li>Cells are made competent with a  <img class="inlinemath" src="eqn005.png" WIDTH=47 HEIGHT=19 STYLE="vertical-align: -4px; margin: 0;" alt="CaCl_2" /> solution</li><li>Commercially available competent cells are usually much more efficient, even though a little bit more expensive</li>
<li>Transfection can be done by heat-shock or electroporation</li>
<li>Plasmids tend to be toxic for bacteria, because of their metabolic burden</li>
<li>DNA absorbs at 260 nm, proteins at 280 nm</li>
<li>Protein biotinylation can be performed in vivo or chemically
<ul>
<li>Chemically, I usually bind biotin to Lys residues</li>
<li>In vivo I can fuse my protein with a biotin binding domain and clone it toghether with BirA, a biotinylating enzyme</li>
</ul></li>
<li>For in silico cloning, we can use Vector NTI Advance or the online platform Benchling</li>
<li>When I create recombinant proteins, it is smart to add a linker
<ul>
<li>Usually Gly is used</li>
</ul></li>
<li>When I clone a protein into an expression vector, I need to optimize codon usage for that organism
<ul>
<li>An old approach was to use the most used codon</li>
<li>Now it is common to respect the codon frequencies of its proteins</li>
<li>When I do codon optimization, I can play a little bit so to avoid or introduce cut sites where needed</li>
</ul></li>
<li>Once I have designed my sequence, I can order it from companies</li>
<li>Addgene is the most used plasmid repository</li>
<li>You can also order a plasmid from your sequence</li>
<li>There are also spin-off that do all the work for you and give you the plasmid to be transfected
<ul>
<li>It is expensive (some thousand â‚¬)</li>
</ul></li>
<li>Bacterial cultures and eukariotic cell lines can be purchased from ATCC (mostly for cell lines) or from DSMZ (Liebniz, for bacteria)
<ul>
<li>Cell lines from ATCC are expensive</li>
<li>Bacterial strains from DSMZ are cheap (20â‚¬)</li>
</ul></li>
<li>Site directed mutagenesis can be used for changing the proprieties of proteins
<ul>
<li>We can change emission wavelenght, increase stability</li>
</ul></li>
</ul>
<h1 id="quantitative-proteomics">Quantitative proteomics</h1>
<ul>
<li>Quantitative proteomics gives us an understanding of the state of the cell</li>
<li>In MS, I can do quantitative analyses with labeled or label-free approaches
<ul>
<li>Labeled approaches are SILAC, ICAT, iTRAQ, O18/O16 enxymatic labeling</li>
<li>In labeled approaches I can run all my samples toghether, while in label-free approaches I can only run one sample at a time</li>
</ul></li>
</ul>
<h2 id="silac">SILAC</h2>
<ul>
<li>The SILAC approach uses a labeled aminoacid in a culture medium deprived of that aminoacid</li>
<li>The aminoacid has to be essential (!)</li>
<li>It is frequent to use deuterated Leu (Leud3)</li>
<li>It can be used only for cell cultures and not for samples like urine, blood, ecc</li>
<li>The 2 samples are mixed early in the sample preparation process and analysed together by LC-MS</li>
<li>The LC treats in the same way the samples</li>
<li>In MS, I can easily distinguish the peaks from the 2 samples, and compare protein expression levels</li>
<li>I can also perform SILAC in whole organisms by using appropriate food (!)
<ul>
<li>This has to be done for more than 1 generation (!)</li>
</ul></li>
</ul>
<h2 id="icat">ICAT</h2>
<ul>
<li>The ICAT approach uses a reactive group that can label an aminoacid side chain</li>
<li>I can use biotin linked with a thiol-specific reactive group (iodoacetamide) through an etilene glycole polimer linker</li>
<li>The linker can be deuterated/normal</li>
<li>Biotin is used for purifing tagged peptides in both normal and deuterated samples, so to minimize error</li>
<li>It can be used only for proteins with accessible Cys, which is much rarer than Leu (SILAC) (!)</li>
</ul>
<h2 id="itraq">iTRAQ</h2>
<ul>
<li>In the iTRAQ appraoch links an isobaric tag to amine groups
<ul>
<li>The tag is composed of a balance group and a reporter group</li>
<li>There are 4 different couples of reporter-balance groups, but the couples have all the same mass</li>
<li>The tag is made so to fragment by CID in MS/MS analises between peptide and balance group, and between balance and reporter group</li>
<li>In the first MS the different tags behave in an identical way</li>
<li>After fragmentation, the tags are quantified to yeld the relative and absolute abundances of proteins</li>
</ul></li>
</ul>
<h2 id="label-free-approaches">Label-free approaches</h2>
<ul>
<li>In label-free approaches I usually compare LC-MS or LC peaks, and then identify the protein by MS/MS (LC-MS/MS or LC/LC-MS/MS)
<ul>
<li>The main problem of this approach is the experimental error between different runs</li>
<li>The large volume of data requires automation</li>
<li>Peak intensity is not-quantitative, but in a specific setting with ESI it was found to be roughly</li>
</ul></li>
<li>Spectral count is another label-free approach where I count the number of peptides identified from my sample
<ul>
<li>The most represented the protein, the more peptides will be identified</li>
<li>It is not so precise (!)</li>
</ul></li>
</ul>
<h2 id="maldi-imaging">MALDI imaging</h2>
<ul>
<li>In 2D MALDI tissue is sliced with a cryostat and a MALDI matrix is applied</li>
<li>We can do a comparative study for different samples (normal vs patological) or study a single sample in order to reconstruct the protein distribution in the tissue</li>
<li>In 3D MALDI I use consecutive sections of tissue slice to get a 3D model of protein distribution in a tissue</li>
<li>It is similar to immunohistochemistry, but without the need for ABs (!)</li>
<li>It can be used not only for proteins, but also other molecules</li>
<li>I can study many different molecules at the same time</li>
<li>The spatial resolution is not as good as with IHC</li>
<li>It tends to be not representative, since not all molecules can be easily ionized</li>
</ul>
<h2 id="seldi">SELDI</h2>
<ul>
<li>Surface enhanced LDI (SELDI) is a variation of MALDI that instead of mixing the sample with the matrix, uses a surface that has biochemical affinity to the target molecule</li>
<li>The surface is functionalised so to be able to bind the target
<ul>
<li>Common functionalizations are CM10 (weakly positive), C6-C12 (hydrophobic), IMAC30 (metal binding), Q10 (anion exchanger), antibodies, DNA, proteins</li>
</ul></li>
<li>After sample application, we can remove contaminant through washing steps and then apply a matrix on it</li>
<li>Some surfaces select for a range of protein with certain proprieties, others for a specific target</li>
</ul>
<h1 id="many-different--omics">Many different -omics</h1>
<ul>
<li>Cellomics is the monitoring of the whole content of single cells
<ul>
<li>Micro-arrayed donut shaped chambers (DSCs) of different volumes can be used for studying single molecules, cells or multicellular arrangements</li>
<li>Each chamber acts as a separate reaction vessel</li>
<li>Single cells can be placed in pL and nL chambers, while fL chambers are for molecular studies</li>
<li>In this way single cells can be studied in a non invasive and time resolved manner using a microscope</li>
<li>Raman spectroscopy can be used for doing label-free cellomics studies</li>
</ul></li>
<li>The secretome is the ensemble of molecules secreted by a cell type
<ul>
<li>The secretome of mesenchimal stem cells (MSCs) can be useful for cartilage regeneration</li>
<li>This terapeutic effect can be achieved by transplanting MSCs, extracellular matrix or secreted molecules</li>
</ul></li>
<li>The regulome is the set of regulatory components in a cell type
<ul>
<li>SCRAT is a software that can analyze single cell regulome data</li>
<li>It is useful for discriminating cell populations and subpopulations</li>
</ul></li>
</ul>
<h1 id="molecular-interactions">Molecular interactions</h1>
<ul>
<li>Protein protein interactions (and interaction of proteins with other molecules) are at the heart of any biological process</li>
<li>We can charachterize an interaction by studying who the interactors are, and which residues partecipate in the interaction</li>
<li>Interactions in a cell type for the interactome</li>
<li>The interactome can be studied with computational or experimental tools
<ul>
<li>Computational tools can be based on structural studies, co-mutation studies</li>
<li>Experimental approaches involve X-ray and NMR spectroscopy, yeast two hybrid, co-immunoprecipitation and other methods based on affinity</li>
</ul></li>
<li>String is a database of experimental and predicted interactions</li>
<li>Some experimental method can be used in vivo, others in vitro</li>
<li>Physical interactions are experimentally determined with a bait-prey approach</li>
<li>Frequently we want to identify the partner of a specific protein
<ul>
<li>In this case we have 1 bait and a library of preys</li>
</ul></li>
</ul>
<h2 id="yeast-two-hybrid">Yeast two-hybrid</h2>
<ul>
<li>The yeast two-hybrid system uses a trascription factor as a reporter</li>
<li>The TF is composed of a DNA binding domain and an activating domain</li>
<li>The 2 domains are separately linked to bait and prey
<ul>
<li>The DB domain is fused to the bait</li>
<li>The AD is fused to the prey</li>
</ul></li>
<li>A reporter gene is expressed if the 2 proteins interact</li>
<li>We build bait and prey plasmids with the recombinant proteins, which are then transfected in yeast</li>
<li>The DB domains used are those of Gal4 or LexA</li>
<li>The AD domains used are those of Gal4, VP16 or B42</li>
<li>The reporter gene can be a survival gene, or beta galactosidase (blue colonies in X-gal)</li>
<li>This system is useful for a firts screening of large cDNA libraries</li>
</ul>
<h2 id="mammalian-two-hybrid-doi10.214497222pf02">Mammalian two-hybrid (doi/10.2144/97222pf02)</h2>
<ul>
<li>The mammalian two-hybrid uses 3 plasmids</li>
<li>A plasmid harbours the BD, one the AD, and a third comprises a minimal TATA box and the reporter gene</li>
<li>It is used for bait and prey proteins that are not appropiately modified post-translationally by yeast</li>
<li>It is useful also for proteins that reside in a specific organelle</li>
<li>Another advantage is that it is faster: the reporter gene typically used give a visible effect in 48h, while with yeast we have to wait for the colonies to grow 3-4 days</li>
<li>The protein have to interact in the nucleus: the plasmids used include a NLS</li>
<li>The most used reporter genes are SEAP, luciferase, and  <img class="inlinemath" src="eqn000.png" WIDTH=12 HEIGHT=19 STYLE="vertical-align: -4px; margin: 0;" alt="\beta" />-galactosidase</li><li>SEAP (secreted alkaline phosphatese) is a secreted protein that can withstand 65Â°C
<ul>
<li>We can degrade the endogenous alkalin phosphatases of the cells by heating them to 65Â°C so to remove background noise</li>
<li>We can detect alkaline phosphatase activity with fluorescence or chemioluminescence methods using the culture medium</li>
<li>It is advantageous because it is secreted, so we do not need to lyse our cells (!)</li>
</ul></li>
<li>Luciferase from Photinus pyralis (north american firefly) is an enzyme that emits in the yellow-green spectrum (560 nm)
<ul>
<li>It converts the carboxylic acid luciferin into luciferin-adenylate</li>
<li>The adenylate is reactive and reacts in the luciferase active site with molecular oxygen producing an excited product</li>
<li>This product rapidly returns to the ground state emitting a photon</li>
</ul></li>
<li>Bacteria galactosidase is a really stable enzyme
<ul>
<li>Its activity can be tested with ONPG, which is hydrolized to onityrophenol</li>
<li>The product can be detected with fluorescent, chemioluminescent and colorimetric methods</li>
</ul></li>
<li>This system is usually used for a refinment of the first screening done with the yeast two-hybrid</li>
<li>It is not suited for the screening of large libraries</li>
</ul>
<h2 id="pull-down-assays">Pull-down assays</h2>
<ul>
<li>The Pull-down assay uses a recombinant bait-tag complex expressed in E. coli</li>
<li>We can immobilize the bait using antibodies anti-tag, or molecules that bind the tag</li>
<li>After incubation of the immobilised bait with the prey, I perform washing steps
<ul>
<li>The incubation can be done with a cell protein extract</li>
</ul></li>
<li>The bait-prey complexes then can be eluted using SDS loading buffer, or a competitive ligand</li>
<li>The resulting eluate is analysed by SDS-PAGE</li>
<li>Interaction can be detected in the gel by doing a WB with Ab against both bait and prey</li>
<li>Another detection approach is to use a radioactive bait with  <img class="inlinemath" src="eqn006.png" WIDTH=25 HEIGHT=17 STYLE="vertical-align: -1px; margin: 0;" alt="S^{35}" /></li><li>Glutathion-S-transferase is a commonly used tag, pulled down with reduced glutathion
<ul>
<li>The formation of a fusion bait with GST has also the advantage of making the chimera more soluble</li>
</ul></li>
<li>Once the complexes have been detected, the interactors can be identified by MS</li>
</ul>
<h2 id="co-immunoprecipitation">Co-immunoprecipitation</h2>
<ul>
<li>The co-immunoprecipitation assay exploits the specificity of an Ab for the bait</li>
<li>The first step is to obtain a cell extract in non-denaturing conditions</li>
<li>The prey bound to the bait co-precipitates</li>
<li>The Ab-bait complexes are immobilised on a protein A or protein G sepharose gel
<ul>
<li>Protein A and protein G are bacterial cell wall proteins that bind Fc <img class="inlinemath" src="eqn007.png" WIDTH=11 HEIGHT=13 STYLE="vertical-align: -4px; margin: 0;" alt="\gamma" /></li></ul></li>
<li>After a washing step, we can elute and then identify our interactors</li>
<li>The main advantage is that I operate with a cell extract, so I do not have alterations due to expression systems and the environment is as similar as possible to the in vivo situation</li>
<li>The identification is similar to the other pull-down assays: I detach the complexes with SDS sample buffer and then identify them with WB</li>
<li>There is the underlying assumption that proteins that co-precipitate with the prey interact with it in a biologically significant way: this has to be proven</li>
</ul>
<h2 id="protein-arrays-and-chips">Protein arrays and chips</h2>
<ul>
<li>Arrays are differentiated from chips because in arrays we have molecules immobilized in high density, while chips exploit microfluidic technologies</li>
<li>Protein arrays can be used to determine interacting partners</li>
<li>An array can be configured in 3 main ways
<ul>
<li>With an immobilised capturing agent and the target in solution</li>
<li>With an immobilised target and a series of putative interactors in solution</li>
<li>With an immobilised complex sample (reverse phase)</li>
</ul></li>
<li>Protein microarrays allow an high throughput analysis of proteins immobilised on a surface
<ul>
<li>We can use protein, nucleic acids, lipids, small molecules as probes</li>
</ul></li>
<li>The protein microarray substrate can be glass, polyacrilamide or agarose gels, nanowells</li>
<li>The immobilisation of proteins can be achieved in different ways
<ul>
<li>By diffusion into the surface</li>
<li>By adsorption</li>
<li>Covalent attachment thorugh specific side chains</li>
<li>By affinity ligands</li>
</ul></li>
<li>Immobilization of nucleic acids on a chip is easy, but with proteins it is more tricky because you can loose functionality
<ul>
<li>The orientation with which the protein binds the surface is the main problem</li>
<li>This can be overcome by using affinity binding, that forces the protein to orient in a specific and modulable way</li>
<li>Covalent binding is usually achieved with Lys residues</li>
</ul></li>
<li>Protein array printing is done by robots, with contanct or non contact methods
<ul>
<li>Actually not only proteins, but any biomolecule can be deposited in this way onto an array</li>
<li>Non contact methods are similar to a traditional inkjet printer
<ul>
<li>Small droplets of probe solution are expelled and dry onto the glass surface of the array</li>
<li>It allows to achieve a slightly higher spot density</li>
<li>It avoids to alter the surface by touching it</li>
</ul></li>
<li>In contact printing, each print pin applies the probe solution onto the surface
<ul>
<li>The printing pin is a needle with a capillary channel inside it</li>
<li>After touching the surface, a small aliquote of solution adheres to the surface</li>
</ul></li>
<li>The main danger of these methods is cross-contamination between the spots</li>
<li>A single spot typically consists of a few nL and covers 100-150  <img class="inlinemath" src="eqn008.png" WIDTH=12 HEIGHT=13 STYLE="vertical-align: -4px; margin: 0;" alt="\mu" />m</li></ul></li>
<li>Expression arrays are used to identify expressed proteins</li>
<li>Interaction arrays are used for studying the interaction parteners of a target</li>
<li>Differential profiling is done by mixing a disease and an healthy sample marked with different Cy dyes
<ul>
<li>I can incubate an Ab array with the mixture, and get information on the different proteins expressed in the different states</li>
<li>Labelling complex samples with Cy dyes can be difficult</li>
</ul></li>
</ul>
<h2 id="aptamers">Aptamers</h2>
<ul>
<li>Aptamers are a syntetic peptides or oligonucleotides that can bind a target in a really specific way because of their 3D structure</li>
<li>They can be used as drugs, delivery mechanisms, in imaging, analytical applications</li>
<li>They are engineered with the SELEX strategy
<ul>
<li>I choose an initial library of nucleic acids/peptides</li>
<li>I incubate the library with the desired target</li>
<li>I perform washing steps so that only aptamers that bind the target are retained</li>
<li>I charachterise the bound aptamers and I amplify them
<ul>
<li>This can be done by PCR for DNA</li>
<li>For proteins I sequence them and then prepare oligonucleotides with the right coding sequence</li>
</ul></li>
<li>In this way I get a new library, and I can repeat the process again</li>
<li>Typically 10-15 rounds are performed</li>
</ul></li>
<li>This is a targeted evolution process</li>
<li>I am selecting for aptamers that interact with the tag in the experimental conditions, but I do not know if they will bind to it also in vivo (!)</li>
<li>Peptide nucleic acids (PNAs) are molecules with an aminoacidic backbone linked by peptide bonds, but have the traditional 4 nucleotides as side chains
<ul>
<li>THe nucleotides are linked not to the C <img class="inlinemath" src="eqn001.png" WIDTH=12 HEIGHT=10 STYLE="vertical-align: -1px; margin: 0;" alt="\alpha" /></li><li>They bind DNA and RNA in a really specific and strong manner, because they lack charge at the level of the backbone and because of their flexibility</li>
</ul></li>
<li>I have to test also for selectivity</li>
<li>It is a really expensive system, and to date only the aptamer for thrombin is really effective</li>
</ul>
<h2 id="phage-display">Phage display</h2>
<ul>
<li>Phage display is an in vitro selection technique that uses a target protein fused with the coat protein of a phage</li>
<li>I can easily build a library of putative interactors to my protein of interest</li>
<li>The selection is done with a standard affinity techniques using the protein/molecule of interest</li>
<li>High affinity phages bind the support, and then can be eluted with free bait</li>
<li>The phages can be amplified by infecting E. coli, and then selected again</li>
<li>Through repetitive selection and infection cycles, we can select for strong binders to the protein of interest</li>
<li>I can use error-prone PCR to mutate my protein and select advantageous mutation for binding to the substrate</li>
<li>Once I am satisfied with the result, I can just sequence the phage and get the corresponding protein sequence</li>
<li>It is used in drug discovery for selecting targets and for selecting high specificity antibodies</li>
<li>It is well-suited for screening large libraries</li>
<li>As always, are we sure that our findings will translate in vivo?</li>
<li>If the first screening is washed too arshly, I can loose some variability</li>
<li>I can only use small peptides, big proteins disrupt the fold of the coat protein</li>
</ul>
<h2 id="fret">FRET</h2>
<ul>
<li>It uses 2 fluorescent molecules, one emitting on the absorption wavelenght of the other
<ul>
<li>There has to be spectral overlap between emission of the donor and absorption of the acceptor</li>
</ul></li>
<li>The energy transfer is non-radiative: the relative orientation of the dipole moment of the molecules must be approximately parallel</li>
<li>The resonace efficiency depends on the inverse 6th power of the distance (generally 1-10 nm is the range)
<ul>
<li>It is really strongly dependent on the distance between the 2 molecules, so it can detect interactions</li>
</ul></li>
<li>RET signal is rationmetric: it does not depend on the number of cells, volume and other variables</li>
<li>The signal is calculated as a ratio of the emissions of acceptor and donor, minus a correction factor that accounts for the spectral overlapping of donor and acceptor
<ul>
<li>A signal of 1 means that the emission of the acceptor is equal to that of the donor</li>
</ul></li>
<li>FRET can be used to show in vivo interactions</li>
<li>BRET is like FRET but uses luciferase as donor molecule</li>
<li>In BRET luminescence is a consequence of a chemical reaction: we do not need to excite
<ul>
<li>There can not be autofluorescent background, and no photobleaching</li>
</ul></li>
<li>BRET is a naturally occurring phenomenon in Acquorea victoria, where GFP is the acceptor</li>
<li>Molecules used in FRET are Cy dyes, fluorescent proteins, FITC, AlexaFluor, Texas Red, Europium compounds</li>
<li>For fluorescent proteins I can just clone a recombinant protein (in vivo!), while for organic molecules I need click chemistry (in vitro!)</li>
<li>FRET can be used for detecting not only interaction, but also conformational changes (!)
<ul>
<li>This is useful for detecting enzimatic activity, intramolecular interactions, ligand binding</li>
<li>I can engeneer a sensor that emits FRET when there is kinase or protease activity in the medium</li>
</ul></li>
<li>The SRET (sequential BRET-FRET) uses a 3-step energy trasfer
<ul>
<li>Its efficiency is quite low (!)</li>
<li>The first step is the transfer of energy from a luciferase to an acceptor protein (GFP), which act as a donor for a final acceptor (YFP)</li>
<li>It has been used for studying GPCRs</li>
</ul></li>
<li>Since we use fusion proteins, we can alterate the structure of our protein (!)</li>
<li>When we use BRET for monitoring homodimerization, I have to consider that I will have many dimers with the same tag (so no fluorescence (!))</li>
<li>we can also use quantum dots as acceptors
<ul>
<li>A quantum dot is a semiconductor nanocrystal of 2-8 nm</li>
<li>Their abosrption spectrum is really wide, it can be excited by almost any fluorescent biomolecule</li>
<li>Its emission can be finely tuned by particle size, and its emission spectrum is really narrow</li>
<li>They have a very large Stokes shift (absorption/emission spectral separation)</li>
</ul></li>
<li>QD-BRET uses a quantum dot bioconjugate with Renilla Luciferase that is able of self-excitation
<ul>
<li>It is advantageous because we can make many differemt QD with really specific spectra, thus we can do multiplexing</li>
</ul></li>
<li>Inteins are peptides capable of excising themselves from proteins, in a process known as protein splicing</li>
<li>QD-BRET was pioneerly used to detect metalloprotease activity in serum
<ul>
<li>The conjugation of QD to RLuc was mediated by an intein phused to RLuc, that catalysed the splicing of RLuc itslef with a QD containing a reactive NH group</li>
<li>Proteases act on the linker between QD and RLuc, impeding BRET to occur</li>
</ul></li>
<li>Modified Inteins can be used for tag-less protein purification
<ul>
<li>A mutated version of an intein is unable to splice itself out at slightly acidic pH</li>
<li>I can link insert the intein between the protein of interest and a tag</li>
<li>After purufucation, I can change the pH and release the native, tagless protein</li>
</ul></li>
</ul>
<h2 id="protein-complementation-assay">Protein complementation assay</h2>
<ul>
<li>Protein complementation assay (PCA) is based on the expression of a luciferase as separated domains bound to proteins that we think could interact</li>
<li>If the proteins interact, the luciferase assembles and we can detect luminescence</li>
<li>PCA can also be used to detect intramolecular interactions</li>
<li>The major drawback is that the 2 halves of the luciferase have high affinity for each other, so they could interact even without interaction of the target proteins (!)</li>
<li>With metil-binding proteins, PCA can be used to asses DNA metilation</li>
<li>The binding of the 2 protein fragments of Luc is really cooperative, therefore it goes from completely undetactable to maximum signal in a really narrow range
<ul>
<li>This is in contrast to FRET, which has a really low dinamyc range and requires extensive optimization</li>
</ul></li>
<li>NanoBiT is a next-generation PCA approach that was develloped by targeted evolution
<ul>
<li>The 2 halves have low affinity for each other, so they do not give an aspecific signal</li>
<li>They have low steric hindrance, so they do not disturb interactions among the proteins of interest</li>
</ul></li>
<li>We can combine PCA and BRET/FRET to assay non-binary interactions
<ul>
<li>The donor or acceptor can be made of 2 halves that are functional only if assembled</li>
</ul></li>
</ul>
<h2 id="in-vivo-imaging">In vivo imaging</h2>
<ul>
<li>We can use these tools also for in vivo imaging</li>
<li>In vivo imaging is in complaiance with the 3R rule
<ul>
<li>Reduction of the number of animals required</li>
<li>Refinement of the experiment so to minimize harm</li>
<li>Replace, if possible, the use of animals</li>
</ul></li>
<li>There is no need to kill the animal, so we can get imaging at different timepoints, minimizing cost and ethical concerns(!)</li>
<li>We can use these tools to monitor the occurrence of specific biological events in live animals</li>
<li>The general method is to overlap a light image of the animal with the signal from the imaging method used
<ul>
<li>Detection is done in a dark box</li>
<li>With the overlapping, we have information on both signal intensity and location</li>
</ul></li>
<li>Bioluminescent imaging (BLI) methods are superior to fluorescent ones because they are not affected by background fluorescence</li>
<li>Some applications of BLI
<ul>
<li>We can engineer pathogenic organisms to emit bioluminescence so to study their localization in vivo, or the activation of specific genes</li>
<li>They can also be used to monitor drug efficacy</li>
<li>We can use a similar strategy to engineer bioluminescent tumor cells</li>
<li>We can detect the survival and replication of stem cells grafts</li>
</ul></li>
<li>The nude mouse is a really useful model because it lacks the thymus and it is nude
<ul>
<li>It is easier to use for imaging, and it readly accepts xenografts</li>
</ul></li>
<li>Traditionally BLI methods employ Photinus pyralis (North american firefly) luciferase, which emits in the green spectrum</li>
<li>Green light is absorbed strongly by mammalian tissues, especially by hemoglobin</li>
<li>It is better to use red light emission spectra, because they penetrate more easily through tissues</li>
<li>NIR-II imaging uses a wavelenght in the second NIR, that minimizes scattering and has an higher tissue penetration
<ul>
<li>NIR-II fluorophores are single walled carbon nanotubes (SWCN), QDs and polymethine dyes</li>
</ul></li>
<li>For QD and SWCN, we have no information about the clinical safety of these devices</li>
</ul>
<h1 id="function-prediction-and-databases">Function prediction and databases</h1>
<ul>
<li>Functional annotation of genomic sequences is a major biological challenge</li>
<li>GOLD (genome online db) is a comprehensive database of genome and metagenome sequencing projects</li>
<li>Metagenomics is the sequencing of environmental DNA
<ul>
<li>Samples from the environment are cloned into vectors and transfected in E. coli in order to build a library</li>
</ul></li>
<li>GO slim is a simplified subset of GO terms used for facilitating browsing</li>
<li>Annotation trasfer is mainly done by homology (sequence similarity)
<ul>
<li>If sequence similarity is higher than 80% we usually have the same function</li>
<li>Sometimes this criterion is misleading so tools tend to integrate other informations on the protein</li>
<li>Most of the newly discovered proteins do not have high similarity with any protein of known function</li>
</ul></li>
<li>Interpro provides a functional charachterization of a sequence
<ul>
<li>It is a meta-database that gets information by many other specialized databases</li>
<li>Some databases are strong in a specific case, others in other cases, and InterPro gets the best of all of them</li>
</ul></li>
<li>BLAST2GO is a tool for functional analysis of sequences
<ul>
<li>It supports EC numbers, KEGG maps, Interpro motives</li>
</ul></li>
<li>KEGG is a collection of manually drawn pathway maps</li>
<li>PANTHER is a classification system for proteins based on evolutionary relationships</li>
<li>MIPS (mammalian protein-protein interaction db) is a manually curated collection of data taken by publications</li>
<li>STRING is a database of known and predicted protein interactions</li>
<li>In protein interactions, we can define an interaction interface
<ul>
<li>Residue are considered interacting when they are less than 0.5  <img class="inlinemath" src="eqn009.png" WIDTH=14 HEIGHT=19 STYLE="vertical-align: -1px; margin: 0;" alt="\AA" /> plus Van der Waals radii apart</li><li>Residues close to each other but not in contact are considered nearby residues</li>
</ul></li>
<li>Interfaceomics is the ensemble of interaction interfaces</li>
<li>PDBePISA is a tool for the exploration of interfaces and prediction of quaternary structures and prediction of quaternary structures</li>
<li>The SKL tag targets proteins to peroxisomes and it is recognised by the PTS1 receptor</li>
<li>When chosing a tool, we should be careful to check if it is still active and maintained</li>
</ul>
<h1 id="poster-presentation">Poster presentation</h1>
<ul>
<li>We should select an original research article according to impact factor of the journal, H-index of the author</li>
<li>The impact factor is really dependent on the field, so we should check the ranking of journal in its subject category</li>
<li>If we choose a paper about a bioinformatics tool, it is better to choose one in a bioinfromatics journal because they tend to be more statistically robust</li>
<li>Standard size is 70*100 cm</li>
<li>Title at the top, with the affiliation logo</li>
<li>Few sentences, aimed at non-specialists</li>
<li>I want collaboration with people in other fields (!)</li>
</ul>
