<h1 id="course-organization">Course organization</h1>
<ul>
<li>Lab1 in the first (Lab1a) and second semester are actually separate courses</li>
<li>In february there is a 1 week intensive course run by professor Allegravia, which is part of Lab1a
<ul>
<li>It is about protein-protein interaction</li>
<li>We will have an exam also for this part</li>
</ul></li>
<li>There will be a written test on 09/01/20</li>
<li>We have to write a report for 09/01/20 or 22/01/20
<ul>
<li>If we submit for the first deadline we will have feedback</li>
</ul></li>
<li>The oral defence will be in the last week of January
<ul>
<li>Probably it will be 29-31/01/20</li>
</ul></li>
</ul>
<h1 id="things-to-have-in-mind">Things to have in mind</h1>
<ul>
<li>H bonds are 3  <img class="inlinemath" src="eqn000.png" WIDTH=14 HEIGHT=19 STYLE="vertical-align: -1px; margin: 0;" alt="\AA" /> long, they require planarity and their E = -4 kcal/mol</li><li>Salt bridge -5 kcal/mol, 4  <img class="inlinemath" src="eqn000.png" WIDTH=14 HEIGHT=19 STYLE="vertical-align: -1px; margin: 0;" alt="\AA" /></li><li>Lennard-Jones interactions have E = -1 kcal/mol</li>
<li>SS bonds -60 kcal/mol, 2.05  <img class="inlinemath" src="eqn000.png" WIDTH=14 HEIGHT=19 STYLE="vertical-align: -1px; margin: 0;" alt="\AA" /></li><li>Covalent bonds 2-3  <img class="inlinemath" src="eqn000.png" WIDTH=14 HEIGHT=19 STYLE="vertical-align: -1px; margin: 0;" alt="\AA" />, -100 kcal/mol</li></ul>
<h1 id="introduction">Introduction</h1>
<ul>
<li>Hydrogens bond are mainly located at the level of the backbone</li>
<li>The reference source in the filed is the Journal of Bioinformatics</li>
<li>Functional annotation is the core of this course</li>
<li>We will focus on proteins because we have many structures available, and we can establish a clear relationship among protein structure and function</li>
<li>Functional annotation studies the relationship between structure and function</li>
<li>Functional annotation requires data collection, storage and analysis</li>
<li>Functional annotation is the activity of attributing structural and functional features to translated protein sequences</li>
<li>Functional annotation is not prediction: we use data to infer properties</li>
<li>Functional genomics is the process of how the genome gives rise to the proteome and metabolome, which in turn influence gene expression</li>
<li>Before starting data analysis, be sure of the quality of your data (!)</li>
<li>A database must be implemented, curated and mined</li>
<li>Database curation refers to updating the data and to keeping them compliant with the database standard</li>
<li>A database release is its content at a given date</li>
<li>Data mining is the retrieval of information from a database
<ul>
<li>It is done with a browser</li>
</ul></li>
<li>A relational database is organized in tables in relation with each other</li>
<li>A file is a container of data</li>
<li>A file format is the specific organization of a collection of data</li>
<li>In Hamburg there is an EMBL faicility that uses X-rays in a flow citometer</li>
<li>A protein is stable because of H-bonding in the backbone</li>
</ul>
<h1 id="x-ray-cristallography">X-ray cristallography</h1>
<ul>
<li>A molecule is carachterized by its electron density</li>
<li>Affinity among molecules comes from the electron densities</li>
<li>A PDB file is not the protein structure, but a way to represent it</li>
<li>A protein is
<ul>
<li>A polymer formed of amminoacil residues joined together by peptidic bonds</li>
<li>A frustrated system capable of auto-organization in the solvent</li>
<li>A social entity that interacts with other molecules</li>
</ul></li>
<li>I can understand that I have a protein crystal by shining light on it and seeing how it diffracts</li>
<li>Routinely X-ray diffraction is not able to locate hydrogens</li>
<li>Each crystal has a unit cell</li>
<li>Electron density is the result of data analysis on diffraction maps</li>
<li>The main cristallographic techniques are NMR and X-rays</li>
<li>X-rays are used because they have a wavelenght comparable to an atomic bond, and therefore they can resolve with that level of detail</li>
<li>X-ray sources used for studying protein structure are rotating anode tubes or synchrotrons</li>
<li>In the rotating anode tube electrons are accelerated towards the anode in a vacuum
<ul>
<li>The anode rotates, so to expose a different portion to the electron beam at each moment and avoid damage</li>
<li>X-rays are releades when the electrons collide with the anode, as a way to get rid of their kinetic energy</li>
<li>Only 1% of the kinetic energy is converted to photons, the rest heats up the anode</li>
</ul></li>
<li>Sychrothron source give an anisotrope beam, which gives semicircles in the diffraction map instead of focused spot
<ul>
<li>They are located in the US, Europe or Japan</li>
<li>X-rays are produced by charged particles moving in a magnetic field</li>
</ul></li>
<li>In order to have a diffraction pattern you need to have interference between the diffracted beams</li>
<li>Bragg diffraction law:  <img class="inlinemath" src="eqn001.png" WIDTH=106 HEIGHT=16 STYLE="vertical-align: -1px; margin: 0;" alt="2d*\sin\theta=n\lambda" /> <ul>
<li>When the same beam is reflected by 2 planes, the part which is reflected by the lower plane travels for a longer distance</li>
<li>The distance is exactly  <img class="inlinemath" src="eqn002.png" WIDTH=65 HEIGHT=16 STYLE="vertical-align: -1px; margin: 0;" alt="2d*sin\theta" />, where  <img class="inlinemath" src="eqn003.png" WIDTH=10 HEIGHT=16 STYLE="vertical-align: -1px; margin: 0;" alt="\theta" /> is the angle of incidence and d is the distance between the planes</li><li>If this distance is equal to n wavelenghts, we observe reflection because of interference</li>
<li>By knowing all the terms except d, I can derive the minimal distance between the diffraction planes</li>
</ul></li>
<li>A typical diffraction maps is organized with 3 coordinates (H, K, L) and an intensity dimension (I)
<ul>
<li>The intesity is proportional to the amplitude, therefore to the amount of costructive interference</li>
<li>The 3 coordinates refer to the reciprocal space of the Fourier transform</li>
<li>I can recove the electron densities from the diffraction pattern with the Fourier transform
<ul>
<li>It is a computation-heavy task</li>
</ul></li>
</ul></li>
<li>I cannot recover the phase from the diffraction map
<ul>
<li>In a synchrotron, I can recover the phase of the wave with the anisotropy approach</li>
</ul></li>
<li>Errors in the electron density mainly derive from errors in the phases</li>
<li>The resolution of a diffraction map depends on how ordered the crystal is
<ul>
<li>It is typically 0.5-5  <img class="inlinemath" src="eqn000.png" WIDTH=14 HEIGHT=19 STYLE="vertical-align: -1px; margin: 0;" alt="\AA" /></li></ul></li>
<li>Once I have the electron density, I need to fit my molecule in it to determin the conformation
<ul>
<li>This is easier if my electron density has an high resolution</li>
<li>I can take advantage of similar proteins with a similar structure to do the fitting</li>
<li>This fitting procedure is called refinement, because it reduces noise in the model</li>
</ul></li>
<li>To validate my model, I compute the diffraction pattern of the theoretical protein structure to check if it matches the experimental pattern within a reasonable tollerance
<ul>
<li>This is the R-value (!)</li>
</ul></li>
<li>In order to facilitate crystallization, protein can be stabilized by crosslinking
<ul>
<li>A crosslinker is a molecule that forms chemical bonds with the protein and stabilizes it</li>
</ul></li>
</ul>
<h1 id="crio-electron-microscopy">Crio-electron microscopy</h1>
<ul>
<li>Crio-electron microscopy gives us a diffraction pattern using an electron beam</li>
<li>It is really useful for really big complexes that cannot be cristallized</li>
<li>Resolution is lower than X-ray diffraction</li>
<li>It allowed to obtain the structure of entire ribosomes</li>
<li>It cannot be used with small proteins that cannot whitstand the electron bombardment</li>
<li>It is possible to capture the protein in different conformations, and so understand how it works
<ul>
<li>It was used by Rubinstein to obtain a movie of V-ATPase (!)</li>
</ul></li>
<li>It is really costly, around 6000â‚¬/day</li>
</ul>
<h1 id="nuclear-magnetic-resonance">Nuclear magnetic resonance</h1>
<ul>
<li>It does not require the crystal, it is performed in an homogeneous really concentrated protein solution</li>
<li>The frequencies used are on the radiowave lenght, so really low energies</li>
<li>NMR measures contacts among atoms in solution, it measure frequence changes when nuclei come close to each other</li>
<li>The data produced is a contact map</li>
<li>The only nuclei considered are mainly O, H, C</li>
<li>From NMR I get many conformation for each molecule, and if I visualize it in a molecular visualization software I see many superimposed structures
<ul>
<li>The core is usually stable and in agreement with X-ray data, while regions with high temperature factor have many configurations</li>
<li>From the many structures I can recover a consensus structure</li>
</ul></li>
</ul>
<h1 id="time-resolved-x-ray-crystallography">Time resolved X-ray crystallography</h1>
<ul>
<li>I use pulsed X-ray instead of a continuous beam</li>
<li>I can observe the conformational changes of the protein between different pulses</li>
</ul>
<h1 id="pdb-file">PDB file</h1>
<ul>
<li>The PDB file does not contain the electron density, it is an approximation of the structure</li>
<li>The resolution of an X-ray diffraction is important
<ul>
<li> <img class="inlinemath" src="eqn004.png" WIDTH=35 HEIGHT=19 STYLE="vertical-align: -1px; margin: 0;" alt="5.0\AA" /> resolution is reasonably accurate only for the position of the backbone</li><li> <img class="inlinemath" src="eqn005.png" WIDTH=35 HEIGHT=19 STYLE="vertical-align: -1px; margin: 0;" alt="1.5\AA" /> can be generally trusted, also for drug design</li></ul></li>
<li>A ligand in PDB is any molecule co-cristallized with the macromolecule considered</li>
<li>An heteroatom is any atom in a PDB file that does not belong to the primary sequenc eof the protein</li>
<li>Signal peptides are 10 to 30 residues in length
<ul>
<li>They are usually cleaved and therefore they do not appear in the protein 3d strucutre</li>
</ul></li>
<li>A PDB file has an unique identifier of 4 letters and numbers</li>
<li>On PDB I can also find the FASTA file for the protein
<ul>
<li>FASTA contains the covalent structure (i.e the sequence) of the protein</li>
<li>FASTA has 60 residues per line</li>
<li>It is the sequence derived from the structure, it can be different than the one in uniprot (!)</li>
</ul></li>
<li>Coverage refers to the percentage of protein sequences covered in the protein structure</li>
<li>PDB files produced using a synchrotron source have 2 spots associated with every atom</li>
<li>For each ATOM we have the xyz coordinates, the occupancy, and temperature factor (B size)</li>
<li>Occupancy is how well the atom fits the electron density
<ul>
<li>It is usually 1</li>
<li>If it is less than 1, there are more records for the same atom with occupancies that add up to 1</li>
<li>The same atom represented more than once has the residue name changed as ARES, BRES, CRES, â€¦</li>
</ul></li>
<li>The temperature factor refers to the mobility of the position
<ul>
<li>Tends to be higher at the surface and lower in the core</li>
</ul></li>
<li>The PDB file contains the atomic model of a macromolecule</li>
<li>The CPK colorscheme is a popular set of colors used for the different atoms</li>
<li>The structure validation window reports the percentile rank of different validation methods
<ul>
<li>Blue is good, red is bad</li>
</ul></li>
<li>DSSP is a program that reads a PDB file and assigns a secondary structure to each PDB coordinate
<ul>
<li>It was made by Sanders, one of the founders of bioinformatics, and Kabsch</li>
</ul></li>
<li>A database can be defined by its statistics</li>
<li>Data in a DB can be distributed in categories that are relevant for the interpretation of data</li>
<li>The space group refers to the simmetries of the unit cell</li>
<li>The Ramachandran plot of a structure can be generated with Procheck (EMBL)
<ul>
<li>It is much more informative than the 3d view generated with Rasmol</li>
</ul></li>
<li>Some PDB statistics
<ul>
<li>158180 macromolecular structures</li>
<li>76380 enzymes</li>
<li>48974 distinc protein sequences</li>
<li>Resolution range from  <img class="inlinemath" src="eqn006.png" WIDTH=44 HEIGHT=19 STYLE="vertical-align: -1px; margin: 0;" alt="&lt;1\AA" /> to  <img class="inlinemath" src="eqn007.png" WIDTH=57 HEIGHT=19 STYLE="vertical-align: -1px; margin: 0;" alt="&gt;4.6\AA" />, with a peak around  <img class="inlinemath" src="eqn008.png" WIDTH=23 HEIGHT=19 STYLE="vertical-align: -1px; margin: 0;" alt="2\AA" /> <ul>
<li>Distribution not normal with a long right tail</li>
</ul></li>
<li>1702 source organinsms</li>
</ul></li>
<li>In protein structure we do not have signal peptides because they are cleaved post-translationally</li>
<li>For every PDB entry, there is a validation report with all the statistics
<ul>
<li>This is summarized by percentile ranks on the PDB webpage</li>
</ul></li>
<li>Atom coordinates have 5 digits, xx.xxx
<ul>
<li>The significance of the digits depends on the resolution of the structure</li>
</ul></li>
<li>mmCIF is a different file format that is used for representing protein structures</li>
</ul>
<h1 id="pdbsum">PDBsum</h1>
<ul>
<li>It is a pictorial database that provides an at-a-glance overview of the contents of each 3D structure deposited in the Protein Data Bank</li>
<li>It is hosted by EMBL-EBI</li>
<li>It shows also the Procheck/Procheck NMR Ramachandran plot for the structure</li>
<li>It shows the biological unit instead of the unit cell</li>
</ul>
<h1 id="pdb101">PDB101</h1>
<ul>
<li>A crystal is composed of the unit cell, that is translationally repeated in the crystal</li>
<li>The unit cell is composed of asymmetric units, that rotated and translated form the unit cell</li>
<li>The asymmetric unit is the unique part of the crystal structure</li>
<li>The biological assembly is the biologically relevant form</li>
<li>Occupancy of an atom is the fraction of times that atom is in the specified position in the crystal
<ul>
<li>The occupancies for an atom always sum to 1, giving the possible alternate conformations</li>
</ul></li>
<li>The R-value is the fit between the theoretical diffraction pattern of the model and the experimental one
<ul>
<li>0 is a perfect fit, 0.63 is a the fit of a random diffraction pattern</li>
</ul></li>
<li>R-free is another statistic that avoids the bias introduced in the refinement step</li>
</ul>
<h1 id="protein-structural-allignment">Protein structural allignment</h1>
<ul>
<li>Rigid superimposition requires the knowledge of at least 3 non-allineated equivalent residues, while structural allignment requires no previous knowledge of equivalent positions</li>
<li>The output of a structural allignment is a set of superimposed 3D coordinates, one for each input structure</li>
<li>A structural allignment implies a corresponding sequence allignment, from which we can calculate sequence identity and similarity
<ul>
<li>Sequence similarity is meaningful only with an underlying structural similarity</li>
<li>Sequence identity is a score between 0 and 1 that gives the number of corresponding residues after the allignment</li>
<li>Sequence similarity is a score between 0 and 1 that gives the number of similar residues after the allignment</li>
<li>Residues are considered similar if they belong to the same chemical class (polar, non polar, cationic, anionic,â€¦)</li>
<li>Structure is conserved more than sequence (!)</li>
</ul></li>
<li>Generally, I keep 1 protein fix as a template, and I try to superimpose the other backbone onto it, allowing the introduction of gaps</li>
<li>There are many different algorithms to do structural allignment</li>
<li>The reduce representation of a protein contains only the  <img class="inlinemath" src="eqn009.png" WIDTH=94 HEIGHT=16 STYLE="vertical-align: -3px; margin: 0;" alt="C-C\alpha-N" /> elements of the backbone<ul>
<li>Structural allignemnt usually only considers the position of the backbone, so it works on the reduced representation of the protein</li>
</ul></li>
<li>After the allignemnt, it is possible to derive various measures of strucutral similarity
<ul>
<li>The simplest metric is the root mean squared deviation (RMSD) among atomic coordinates
<ul>
<li>It should be below 3  <img class="inlinemath" src="eqn000.png" WIDTH=14 HEIGHT=19 STYLE="vertical-align: -1px; margin: 0;" alt="\AA" /></li></ul></li>
</ul></li>
<li>The raw score (a dimensionless metric) can be normalized by subtracting the mean and dividing by the standard deviation, so to get the z-score</li>
<li>One of the most famous structural allignment alogorithms is jCE (Java Combinatorial Extension), written by Philippe Bourne, the director of the PDB
<ul>
<li>It is one of the best-performing algorithms</li>
<li>It breaks down the proteins in fragments, and it tries to allign the structure of these by several methods (RMSD, secondary structure, â€¦)</li>
<li>It forms a series of alligned fragment pairs (AFPs) and filters them, retaining only those that respect a given measure of local similarity</li>
<li>It generates an optimal path among AFPs, that yelds the final allignment</li>
<li>The first AFP that nucleates the allignment can occur at any position</li>
<li>The size of AFP and the maxium allowed gap are parameters, usually set to heuristic optimal values</li>
<li>An important drawback is that It does not deal well with flexible regions that can have different conformations, since it is based on rigid superimpoisition</li>
</ul></li>
<li>FATCAT is another algorithm that deals better with flexible regions, but can also give spuorios allignments among unrelated regions</li>
<li>Many algorthms cannot recognize structural similarities that are not sequence order dependent</li>
<li>Triangle Match deals with sequence order independent relationships</li>
<li>The lenght of an allignment is the lenght of the protein sequence, plus the gaps introduced</li>
<li>In the PDB, all possible pairwise structural allignments are pre-calculated and stored in xml files
<ul>
<li>The database is updated weekly</li>
</ul></li>
<li>When alligning structures, it is better to use structures taken with the same method</li>
</ul>
<h1 id="protein-structural-classification">Protein structural classification</h1>
<ul>
<li>A protein family is the set of proteins that perform the same function in different organism, and therefore share a similar structure</li>
<li>Protein classification is generally conservative: when in doubt we make a new division</li>
<li>Protein families were discovered by M. Dayhoff</li>
<li>Multiple structural allignment allows to define protein families</li>
<li>We know around 14000 protein families</li>
<li>Multiple allignments are the result of repeated pairwise allignments</li>
<li>We are doing multiple alignments in order to build a sequence profile</li>
<li>A sequence profile is a matrix with residues in the y axis and the position in the alignment in the x axis
<ul>
<li>It is a compressed way to describe a consensus sequence</li>
</ul></li>
<li>A protein family is a set of proteins charachterised by structural superimposition</li>
<li>Protein families are important because they allow us to cluster PDB data
<ul>
<li>Proteins in the same family typically have &gt;30% identity, but there are exceptions (globins)</li>
<li>They are constructed starting by comparing proteins with the same function</li>
<li>It can be then computationally described with structural allignment</li>
<li>A protein family is described with an HMM (hidden markow model)</li>
<li>Hidden Markow Models are also called Pfam domains</li>
</ul></li>
<li>Proteins in the same family can also be really different in sequence, but their structure is really similar</li>
<li>We cannot detect sequence similarity when under 30%, but we can detect structure similarity in those cases
<ul>
<li>Under 30% the result of a sequence allignment is not statistically significant</li>
</ul></li>
<li>Pfam categorizes all the entries in the PDB in protein families, clustering for strucutral similarity
<ul>
<li>The Pfam database was built by performing pairwise comparison of all the PDB entries</li>
<li>A protein family is described by a Hidden Markow Model (HMM)</li>
</ul></li>
<li>A superfamily is a set of protein families with possibly different foldings that can perform the same function
<ul>
<li>The functional similarity suggest common origin, but this is not certain</li>
</ul></li>
<li>A protein domain coincides with the folded protein for small globular proteins (150 aa)</li>
<li>When the PDB grew, we realised that multi-domain proteins share domains with small globular proteins</li>
<li>A protein domain is a protion of the sequence that harbours a function
<ul>
<li>It is also defined as indipendently folding, but are we sure they did experiments about it?</li>
</ul></li>
<li>SCOP categorizes proteins in superfamilies, Pfam families and fold</li>
<li>The SCOP fold can be all alpha, all beta, alpha+beta, alpha/beta, small proteins
<ul>
<li>Alpha+beta has distinct alpha and beta regions</li>
<li>Alpha/beta has mixed alpha-beta structures</li>
</ul></li>
<li>Proteins in the same family have clear common evolutionary origin, and usually have &gt;30% sequence identity</li>
<li>Proteins in the same superfamily have low sequence identity, but common structural and functional features suggest evolutionary relationships</li>
<li>Proteins are said to have the same fold if they have the same secodary structres in the same arrangement and with the same topology
<ul>
<li>If 2 proteins have the same fold they do not need to be evolutionary related: it can be a case of converging evolution</li>
</ul></li>
<li>Margaret Dayhoff studied cytochrome c and determined that it is similar in many organisms
<ul>
<li>She was the first to relate structure and function</li>
</ul></li>
<li>Protein in the same family with less than 30% homology are called distantly related homologs</li>
</ul>
<h1 id="sequence-allignment">Sequence allignment</h1>
<ul>
<li>It is our only way to compare proteins for which I do not have structures</li>
<li>I have around 180,000,000 proteins in UniProtKB
<ul>
<li>99% are only predicted</li>
<li>561,568 in Swiss-Prot</li>
<li>179,250,561 in TrEMBL</li>
</ul></li>
<li>There are at least 3 orders of magnitude among the number of strucutres and sequences in databases
<ul>
<li>Actually more because PDB is redundant and UniProt no</li>
</ul></li>
<li>A gene is a transcribed locus</li>
<li>A sequence allignment is a continuous stretch of residues of any lenght</li>
<li>Sequence comparison can be pairwise or database search</li>
<li>Database search is an extension of pairwise sequence alignment</li>
<li>Sequence allignment can be local or global</li>
<li>Multiple alignments are based on many pairwise alignments</li>
<li>A global allignment optimizes pairing over the whole sequences by introducing gaps
<ul>
<li>A global allignment has a lenght that is at least as long as the longest sequence</li>
</ul></li>
<li>A local alignment stops the allignment if continuing it makes its score lower
<ul>
<li>From a pairwise comparison, I can get many local allignments</li>
</ul></li>
<li>A metric is a set of rules that allow us to define the distance between strings</li>
<li>The Hamming distance is, for a pair of sequences equal in lenght, the number of mismatching positions
<ul>
<li>It is used for ungapped allignments</li>
</ul></li>
<li>The Levensthein or edit distance of 2 strings is the minimal number of edits necessary to change 1 string into the other
<ul>
<li>It is suitable for gapped allignments</li>
<li>An edit operation is defined as delition, insertion or alteration of a single charachter</li>
</ul></li>
<li>A scoring scheme is a measure of sequence similarity
<ul>
<li>It is a substitution matrix where each possible substitution has a score</li>
<li>The matrix is symmetric, so it is often reported only half of it</li>
</ul></li>
<li>Sequence alligment algorthms seek to maximize a scoring function or minimize a dissimilarity measure</li>
<li>For nucleic acids, there are substitution matrices that only consider match vs mismatch, and matrices that give different scores to transitions and transversions</li>
<li>For aminoacids, we have the PAM and BLOSUM matrices, and matrices derived from structure allignment</li>
<li>The PAM matrices were developped by M. Dayhoff and are based on the observed frequencies of mutation of 1 aa into another in alligned proteins of the same family
<ul>
<li>1 PAM is 1% accept mutation, so 2 sequences 1 PAM apart have 99% sequence identity</li>
<li>The matrices were built using closely related sequences 1 PAM apart, so that multiple substitutions were unlikely</li>
</ul></li>
<li>The PAM1 matrix was built by collecting statistics on substitution frequencies in pairwise comparison of sequences 1 PAM apart and correcting for relative aminoacid abundance
<ul>
<li>The score of the mutation i-&gt;j is the log-odd of the mutation</li>
<li> <img class="inlinemath" src="eqn010.png" WIDTH=107 HEIGHT=28 STYLE="vertical-align: -10px; margin: 0;" alt="S=\log{\frac{p(i,j)}{p(i)*p(j)}}" /></li><li>p(i,j) is the observed i-&gt;j mutation rate while p(i) and p(j) are the relative aminoacid abundances</li>
<li>Note that p(i)*p(j) is the expected mutation rate if all mutations are equally likely, it is a correction factor for aminoacid frequencies</li>
<li>Since the score is a really small number, it is usually multiplied by 10</li>
</ul></li>
<li>Other PAM matrices are built as powers of PAM1</li>
<li>PAM250 is used for comparing sequences with 20% identity</li>
<li>Conservation is always positively score, but with different scores depending on aa abundance</li>
<li>The BLOSUM are a family of matrices that also use the log-odds for the substitutions
<ul>
<li>They were produced in the 1990, where there where many more sequences available</li>
<li>They are based on ungapped multiple alignments in short regions of related sequences</li>
<li>The different matrices were built using allignments with different thresholds of sequence identity</li>
<li>Lower matrices are more permissive since are built with sequences that have less than a treshold of sequence identity</li>
</ul></li>
<li>A dotplot is a plot that gives an overview of the similarity between 2 sequences
<ul>
<li>It is also based on scoring schemes</li>
<li>Dotlet is a Java tool for dotplot analisys</li>
<li>It is useful for finding repeated portions and for finding intron-exon boundaries</li>
</ul></li>
<li>Dynamic programming optimizes the solution of subproblems in order to find a global solution
<ul>
<li>It gives the correct solution provided that all the subproblems are independent, but it is computationally expensive</li>
</ul></li>
<li>Dynamic programming approaches are Needelman-Wunsch (global) and Smith-Waterman (local)</li>
<li>Global alligment methods (NW) optimize the allignment over the whole sequence, and can include low-similarity regions</li>
<li>Local allignment methods (SW) can yeld multiple allignments from a single comparison
<ul>
<li>Low similarity regions do not affect the allignment score</li>
<li>Local allignment is preferred in DB searches</li>
</ul></li>
<li>For database searches, we use methods based on words (K-tuples), also called heuristic
<ul>
<li>Heuristic means approximate, it does not give an optimal solution</li>
<li>It also means empirical, not based on theory</li>
</ul></li>
<li>BLAST is an heuristic local allignment method used for database search
<ul>
<li>Originally described by Altschull in 1990 in J. Mol. Bio.</li>
<li>It is 1 order of magnitude faster than other heuristic methods</li>
</ul></li>
<li>FASTA is another heuristic algorthm but is no longer used</li>
<li>An heuristic method is optimized for the expected result, therefore it does not have any intrinsic validity</li>
<li>An expected result is the result of experimental approaches, which is well accepted in the scientific community
<ul>
<li>It is high quality data</li>
</ul></li>
<li>A heuristic method is not based on theory, while QED is firmly based on theoretical ground</li>
<li>Whatever is heuristic is at the core data-driven</li>
<li>In BLAST I chop the query in K-tuples and make a list of words, that I use to scan the DB
<ul>
<li>Word and k-tuples are the same thing</li>
<li>A k-tuple is an ordered set of k values</li>
<li>At this phase BLAST searches for exact matches of words in the list with DB entries</li>
<li>Any of these local alignments can form a maximal segment pair</li>
<li>A maximal segment pair (MSP) is defined as the highest scoring pair of identical segments chosen from 2 sequences
<ul>
<li>It can be of any lenght, so to maximize the score</li>
<li>It provides a measure of local similarity</li>
</ul></li>
<li>In biology we care for all conserved regions, not only the best scoring one
<ul>
<li>To take care of this, a segment pair is defined as locally MSP if its score cannot be improved by extending or shortening both segments</li>
</ul></li>
<li>BLAST filters for all local MSP that score above a cutoff</li>
<li>I want to retrive from a database all the sequences with MSP score above a cutoff T</li>
<li>The greatest advantage of MSP is that we have the matematical tools to determine its statistical significance</li>
</ul></li>
<li>In BLAST, sequences that score far above the cutoff are almost definitely biologically relevant, while borderline matches can be evalued considering the biological context</li>
<li>The behaviour of BLAST can be tweaked with some parameters
<ul>
<li>I can search for exact matches or allow for gaps</li>
<li>I can choose scoring matrice and gap penalty</li>
</ul></li>
<li>BLAST speeds up DB search by avoiding to spend time in sequences that are unlikely to give high MSP scores
<ul>
<li>Given a fixed word lenght w, BLAST seeks only segment pairs with a word of score at least T</li>
<li>When a matche is found, BLAST tries to extend the segment to see if it reaches the desired final cutoff score S</li>
<li>The lower the value of T, the more probable that a segment of score &gt;S will contain a word with score &gt;T</li>
<li>However, the lower the value of T the higher the number of hits, and therefore the execution time</li>
<li>Random simulations allowed to determine an optimal T value for various conditions</li>
</ul></li>
<li>The algorithm first makes a list of words that score &gt;T when compared with some word of the query
<ul>
<li>The time of list generation is linearly proportional to the lenght of the query</li>
</ul></li>
<li>BLAST then tries to extend the MSPs in both directions</li>
<li>During the extension phase, if the score falls below a certain treshold below the score of the original MSP, it is discarded
<ul>
<li>It loses in accuracy, but in a negligible manner</li>
</ul></li>
<li>Theoretical results on the distribution of MSP scores of random sequences allow the following determiantion
<ul>
<li>Given a set of probabilities for the occurrence of each residue and a scoring matrix</li>
<li>The theory gives the parameters  <img class="inlinemath" src="eqn011.png" WIDTH=11 HEIGHT=16 STYLE="vertical-align: -1px; margin: 0;" alt="\lambda" /> and K for evaluating the statistical significance of MSP scores</li><li>With 2 random sequences of lenght m and n, the probability of finding an MSP with score equal or better than S is  <img class="inlinemath" src="eqn012.png" WIDTH=54 HEIGHT=17 STYLE="vertical-align: -3px; margin: 0;" alt="1-e^{-y}" />, with  <img class="inlinemath" src="eqn013.png" WIDTH=150 HEIGHT=20 STYLE="vertical-align: -4px; margin: 0;" alt="y = K*m*n*e^{-\lambda S}" /></li><li>In a similar way, we can calculating the probability of having c MSPs with score greater than S</li>
<li>This result is the p-value of the MSP score</li>
</ul></li>
<li>A sequence allignment method uses its algorithm and substitution matrices to give a result that maximizes the score of the allignment</li>
<li>Sequence allignment methods are less stable than structural ones, more sensitive to lenght of the sequences and other variables</li>
<li>The raw score of a sequence alignment is the sum over its lenght of the score for each match
<ul>
<li>It uses a score substitution matrix to determine the score of each match</li>
<li>It can be demonstrated that raw score follows an extreme value distribution</li>
</ul></li>
<li>The bit-score is the Log scaled version of the raw score
<ul>
<li>It is measured in bit, and it is a metric for the search space</li>
<li>Each unitary increse in bit score doubles the search space</li>
<li>A bit score of 30 means that we expect that score to be observed once in  <img class="inlinemath" src="eqn014.png" WIDTH=25 HEIGHT=17 STYLE="vertical-align: -1px; margin: 0;" alt="2^30" /> comparisons</li><li>It is used by BLAST and it uses a formula that is a bit complex
<ul>
<li> <img class="inlinemath" src="eqn015.png" WIDTH=95 HEIGHT=28 STYLE="vertical-align: -10px; margin: 0;" alt="S'=\frac{\lambda S-\ln(K)}{\ln(2)}" /></li></ul></li>
<li>The bit-score Sâ€™ depends on the parameters  <img class="inlinemath" src="eqn011.png" WIDTH=11 HEIGHT=16 STYLE="vertical-align: -1px; margin: 0;" alt="\lambda" /> and K</li><li>The 2 parameters depend on the substitution matrix and on the gap penalty, and on the size of query and database</li>
<li>It is independent on the size of the search space (dimension of the database), beacuse it corrects for it</li>
</ul></li>
<li>The E-value is a correction of the p-value for multiple testing
<ul>
<li>It is the expected number of matches of that score that I expect in a random database</li>
<li>It depends on  <img class="inlinemath" src="eqn016.png" WIDTH=33 HEIGHT=19 STYLE="vertical-align: -4px; margin: 0;" alt="K, \lambda" /> and the size of the database</li></ul></li>
<li>MegaBLAST is an implementation of BLAST optimized for very long and very similar sequences, such as those differing only for sequencing errors
<ul>
<li>It uses a greedy algorithm</li>
</ul></li>
<li>PsiBLAST (position specific iterated)
<ul>
<li>It takes a single protein sequence as input, and compares it to a protein DB with a normal BLAST search</li>
<li>Given a threshold, it builts a multiple allignment with all the local allignments above the threshold</li>
<li>From the multiple allignment, a profile is built for any local alignment using the query as a base</li>
<li>The profile has the same lenght as the query</li>
<li>The profile is used again for DB searches using a slight modification of the BLAST algorithm</li>
<li>The statistical theory developped for BLAST is also valid in profile searches</li>
<li>The algorithm then iterates the process by building another profile from the allignment of the new hits</li>
<li>The process is repeated a fixed number of times, or until convergence</li>
</ul></li>
</ul>
<h1 id="distantly-related-homologs-30-sequence-identity">Distantly related homologs (30% sequence identity)</h1>
<ul>
<li>Burkhard Rost is a professor in Munich who first published a graph showing the confidence in sequence allignment as sequence identity against number of residues aligned
<ul>
<li>In this graph, it marks the region where we are confident to have evolutionary relationships</li>
<li>The line is the best fit deriving from the data points of structural allignments of all the PDB structures</li>
<li>The region below the best fit is where I cannot be confident that the allignment reflects a structural relationship</li>
<li>The horizontal asymptote is around 30% sequence identity +- something, so irrespective of sequence lenght, under 30% identity we cannot imply strucutural similarity</li>
<li>read paper!</li>
</ul></li>
<li>Also on a statistical standpoint, the allignment is not significant under 30% indentity</li>
<li>Sequence allignment methods are reliable and will give a similar result to structural allignment only when sequence identity is above 30%</li>
<li>Distantly related homologs are proteins that have the same folding and perform the same function, but have a really different sequence
<ul>
<li>We cannot do sequence allignment with them (!)</li>
<li>We can use the multiple sequence allignment (derived from structure) and the HMM of a protein family to model them (!)</li>
</ul></li>
<li>We can have proteins that have the same domains but shuffled in a different order
<ul>
<li>In this case structural allignment is problematic</li>
</ul></li>
</ul>
<h1 id="ramachandran-plot">Ramachandran plot</h1>
<ul>
<li> <img class="inlinemath" src="eqn017.png" WIDTH=12 HEIGHT=10 STYLE="vertical-align: -1px; margin: 0;" alt="\alpha" /> carbons in proteins are  <img class="inlinemath" src="eqn018.png" WIDTH=35 HEIGHT=19 STYLE="vertical-align: -1px; margin: 0;" alt="3.8\AA" /> apart</li><li>The  <img class="inlinemath" src="eqn019.png" WIDTH=12 HEIGHT=19 STYLE="vertical-align: -4px; margin: 0;" alt="\phi" /> angle is the dihedral angle between N-C <img class="inlinemath" src="eqn017.png" WIDTH=12 HEIGHT=10 STYLE="vertical-align: -1px; margin: 0;" alt="\alpha" />,  <img class="inlinemath" src="eqn020.png" WIDTH=13 HEIGHT=19 STYLE="vertical-align: -4px; margin: 0;" alt="\psi" /> is between C <img class="inlinemath" src="eqn017.png" WIDTH=12 HEIGHT=10 STYLE="vertical-align: -1px; margin: 0;" alt="\alpha" />-COOH</li><li>The Ramachandran plot graphs the  <img class="inlinemath" src="eqn019.png" WIDTH=12 HEIGHT=19 STYLE="vertical-align: -4px; margin: 0;" alt="\phi" /> angle of a residue against its  <img class="inlinemath" src="eqn020.png" WIDTH=13 HEIGHT=19 STYLE="vertical-align: -4px; margin: 0;" alt="\psi" /> angle</li><li>Some regions of the plot are really common and allowed, some are not because of steric hindrance</li>
<li>The Ramachandran plot of a protein is a scatterplot of its dihedral angles superimposed on a color code for the allowed conformational spaces</li>
<li>Procheck and Procheck NMR calculate Ramachandran plots from PDB files</li>
<li>A Ramachandran plot is a bidimensional map of a protein structure where the torsion angles of the backbone are reported
<ul>
<li>Donâ€™t say residues, they are in the backbone</li>
</ul></li>
<li>The expected values are determined by measuring torsion angles from a set of well charachterized proteins</li>
<li>The main regions are alpha (A), beta (B), 3-10 helices and left-handed helices (l), and proline region
<ul>
<li>In the top left quadrant (negative  <img class="inlinemath" src="eqn019.png" WIDTH=12 HEIGHT=19 STYLE="vertical-align: -4px; margin: 0;" alt="\phi" />, positive  <img class="inlinemath" src="eqn020.png" WIDTH=13 HEIGHT=19 STYLE="vertical-align: -4px; margin: 0;" alt="\psi" />) we have beta-strands<ul>
<li>Note that beta-strands can be also partially allowed in the extreme bottom left because the angles are circular (180 = -180 !)</li>
<li>Same thing in the extrem top right and extreme bottom right</li>
</ul></li>
<li>Alpha-helices are on the left, vertically centered but more towards the bottom quadrant (negative  <img class="inlinemath" src="eqn019.png" WIDTH=12 HEIGHT=19 STYLE="vertical-align: -4px; margin: 0;" alt="\phi" />, negative to slightly positive  <img class="inlinemath" src="eqn020.png" WIDTH=13 HEIGHT=19 STYLE="vertical-align: -4px; margin: 0;" alt="\psi" />)</li><li>In the top right quadrant we have left-handed 3-10 helices (slightly positive  <img class="inlinemath" src="eqn019.png" WIDTH=12 HEIGHT=19 STYLE="vertical-align: -4px; margin: 0;" alt="\phi" /> and  <img class="inlinemath" src="eqn020.png" WIDTH=13 HEIGHT=19 STYLE="vertical-align: -4px; margin: 0;" alt="\psi" />)</li><li>Proline is special because of its cis peptide bond and has a specific area on the rigth, at the extreme bottom (very negative  <img class="inlinemath" src="eqn020.png" WIDTH=13 HEIGHT=19 STYLE="vertical-align: -4px; margin: 0;" alt="\psi" />, positive  <img class="inlinemath" src="eqn019.png" WIDTH=12 HEIGHT=19 STYLE="vertical-align: -4px; margin: 0;" alt="\phi" />)</li><li>Glycine has really low steric hindrance and can be practically everywhere in the plot</li>
</ul></li>
<li>A good model has at lest 90% of the residues in the most allowed regions
<ul>
<li>This is based on the analysis of 118 structures of at least 2  <img class="inlinemath" src="eqn000.png" WIDTH=14 HEIGHT=19 STYLE="vertical-align: -1px; margin: 0;" alt="\AA" /> resolution and R-factor less than 20%</li></ul></li>
<li>The G factor, for the different angles, measures how unusual a structure is
<ul>
<li>It is a log-odds based on the observed distribution of stereochemical properties</li>
</ul></li>
</ul>
<h1 id="protein-structural-prediction">Protein structural prediction</h1>
<ul>
<li>The goodness of a protein strucuture can be determined by comparison with a set of optimal conditions, determined by analysis of the PDB database</li>
<li>The strongest interaction in proteins are H bonds
<ul>
<li>The electronegative atoms that partecipate in H bonds are O and N in proteins</li>
<li>The main source of H bonding interactions is the backbone</li>
<li>The length of an H bond is around  <img class="inlinemath" src="eqn008.png" WIDTH=23 HEIGHT=19 STYLE="vertical-align: -1px; margin: 0;" alt="2\AA" /> between the 2 electronegative atoms</li><li>The bond can happen only if the atoms are in plane</li>
<li>The strenght of an H bond is around 10 Kcal/mol</li>
</ul></li>
<li>Charge-charge particles are really dependent on the environment</li>
<li>Lennard Jones interactions have an energy around 1 Kcal/mol
<ul>
<li>They are described by the 6-12 potential, since they have a repulsive term ( <img class="inlinemath" src="eqn021.png" WIDTH=29 HEIGHT=17 STYLE="vertical-align: -1px; margin: 0;" alt="R^12" />) and an attractive term ( <img class="inlinemath" src="eqn022.png" WIDTH=20 HEIGHT=17 STYLE="vertical-align: -1px; margin: 0;" alt="R^6" />)</li></ul></li>
<li>SS bonds have energy of 30-40 Kcal/mol and a lenght of  <img class="inlinemath" src="eqn008.png" WIDTH=23 HEIGHT=19 STYLE="vertical-align: -1px; margin: 0;" alt="2\AA" /> <ul>
<li>Their presence depends on the redox ambient potential</li>
</ul></li>
<li>A protein family is charachterized by different sets of GEO terms</li>
<li>Given an unknown sequence, I can allign it to different protein families and eventually translate the GEO terms of the family to it
<ul>
<li>I cannot go below a treshold of 30% of sequence identity with the family template</li>
</ul></li>
<li>It is one of the main shortcut used for predicting the folding of a protein sequence</li>
<li>If in my model I have even a small difference in the active site from that of the family, the GEO terms cannot be traslated</li>
<li>If we donâ€™t have the structure of the protein and we cannot assign it to a protein family, we donâ€™t know anythong about it</li>
<li>If we do have the structure, we can try to compute the function by programs of theoretical chemistry</li>
<li>Pfam is a DB of protein families that should be used for those sequences that donâ€™t find a template in the PDB with sequence allignment</li>
<li>Protein folding usually starts with seeds of alpha-helices that then elongate and promote the final folding</li>
<li>We have 3 methods for computing protein structure
<ul>
<li>Building by homology</li>
<li>Threading</li>
<li>Ab initio</li>
</ul></li>
<li>Which method is better to use depends on the availability of strucutures that have a sequence identity over a threshold with my sequence
<ul>
<li>Above 30% I can use the concept of protein family (building by homology)</li>
<li>Below 30% I need to use threading, fold recognition, machine learning</li>
<li>If I have a new folding, I need ab initio or machine learning</li>
</ul></li>
<li>If I get a model, I can understand its validity by checking if it would be stable</li>
<li>Comparative modelling is a procedure that, starting from a sequence
<ul>
<li>Selects a template by alligning (BLAST) against all the PDB structures</li>
<li>Once found a good template, I can use NW (global allignment) to improve my allignment with the target</li>
<li>I then use modeller for modelling my sequence on the template</li>
<li>The output of modeller is a PDB file containing the model</li>
<li>I check with Procheck the plausibility of my model</li>
<li>The goodness of my model depends on the goodness of the initial allignment</li>
<li>If I am not satisfied I allign again and model again, until I am satisfied</li>
</ul></li>
<li>Threading procedures are based on the modelling of my sequence on different folds
<ul>
<li>In the PDB we have folds, that are self-stable and can be astracted from the protein</li>
<li>The main difference among different folds is the secondary structure, hence the topology of my model
<ul>
<li>In essence, the difference is in the pattern of H bonds</li>
</ul></li>
<li>I have a scoring function that considers the stability of my model (H bonds, other interactions)
<ul>
<li>I get a likelyhood value for every model</li>
</ul></li>
<li>I select the best model among the computed ones</li>
<li>Different models can better cover different portions of my sequence</li>
</ul></li>
<li>Ab initio minimizes the energy of the system by computing all the pairwise interactions
<ul>
<li>It is feasible only on small sequences</li>
</ul></li>
<li>There is a paper written by the author of the Rosetta method (Sanchez, 2000) that analised the validity of the different methods
<ul>
<li>If my starting sequence identity is above 60% my model is good for docking and it is comparable to a low quality experimental structure</li>
<li>In the 60-30% range we have a rough idea of the organisation of the backbone</li>
<li>Under 30% we can be lucky, or completely wrong</li>
</ul></li>
<li>The coverage of a local allignment is its lenght compared to the sequences</li>
<li>How to select the best template
<ul>
<li>Highest sequence identity</li>
<li>Highest coverage, at least 70%</li>
<li>Highest template resolution</li>
</ul></li>
<li>Modeller was written by Sali, a PhD student, who became really famous
<ul>
<li>It is written in Fortran</li>
<li>It transplants the coordinates of the template to the target and it seeks for protein stability</li>
<li>It checks if the pairwise interactions are conserved (spatial restraints)</li>
</ul></li>
<li>Once we get the model, if it does not make sense we can try to tweak the initial sequence allignment</li>
<li>If the model respects the Ramachandran plot, we can allign it to the template structure</li>
<li>When trasferring functional annotation, we have to be careful of the meaning of what we are doing
<ul>
<li>The GO Cellular component can be different even if the structure is conserved</li>
</ul></li>
<li>Swiss-model runs modeller in remote, and it has many pre-computed models</li>
<li>We need to be carefull with the pre-calculated models because they do not have any quality check (!)</li>
<li>Modeller is a software that models the 3D structure of proteins by satisfaction of spatial restraints</li>
<li>The input for modeller are the set of spatial restraints on the structure of the protein and of the ligand to be modelled</li>
<li>The output is a structure for the target that satisfies the restraints as well as possible
<ul>
<li>Restraints are distances, angles, pairs of angles, ecc.</li>
<li>The restraints are automatically derived from the allignment with the template</li>
<li>A restraint is defined in term of a probability density function</li>
</ul></li>
<li>In modeller I can ask as many models as I want, and they are scored from best to worse</li>
</ul>
<h1 id="modeller-from-manual-and-other-source">Modeller (from manual and other source)</h1>
<ul>
<li>Modeller is a computer program that models three-dimensional structures of proteins and their assemblies by satisfaction of spatial restraints</li>
<li>The input to the program are the restraints on the spatial structure of the aminoacid sequence to be modeled</li>
<li>The output is a 3d structure that satisfies the restraint as well as possible</li>
<li>Restraints can be related structures, NMR experimental data, rules of secondary structure packing, other experiments</li>
<li>The model is computed by optimization of a the modeller objective function called molpdf</li>
<li>The sequence alignment of target and template must be given as input</li>
<li>Given an alignment, the restraints on distances, angles and other features are automatically derived from the staistical analisys of the relationships between many pairs of homologous structures
<ul>
<li>This was based on 105 families that included 416 proteins of known structure</li>
</ul></li>
<li>From the statistical distribution of a restraint (e.g.Â distance of related C-C bonds) I can derive the pdf for that specifc property</li>
<li>Spatial restraints and CHARM energy terms are combined into an objective function</li>
<li>The objective function is minimized in a Cartesian space via conjugate gradients, molecular dynamics and simulated annealing</li>
<li>Different models can be obtained by varying the initial structure
<ul>
<li>This is done in a randomized way</li>
</ul></li>
<li>Variability among models can be interpreted as error in a specific region of the fold</li>
<li>Some regions use a specialized modeling protocol (e.g.Â loops)</li>
<li>The optimization is iterated, first satisfying short-range restraint and then long-range</li>
<li>Every model produced is carachterized by molpdf, DOPE score and GA341 score</li>
<li>The DOPE (discrete optimized protein energy) is a statistical potential described by Sali in a paper (doi: 10.1110/ps.062416606)
<ul>
<li>It approximates the free energy of a protein, since the native conformation was demonstrated to have the lowest free energy (Anfinsen, 1972)</li>
<li>It is too costly to compute the free energy</li>
<li>DOPE is constructed from a set of crystallographic structures, so it is knowledge-based</li>
</ul></li>
<li>GA341 (doi: 10.1110/ps.062095806) is a function of the statistical potentials, z-score, compactness and sequence identity with the template
<ul>
<li>It ranges from 0 to 1, where 1 is a good model</li>
<li>It is based on machine learning</li>
</ul></li>
</ul>
